Updated docs with passkey limitation information (see context section in description) (#2291)
clerk/clerk-docs: docs/_partials/authentication/passkeys-restriction.mdx

Passkeys are tied to the domain they are created on and **cannot be used across different domains**. For example:
Passkeys are tied to the domain they are created on and **cannot be used across different domains**. However, passkeys **do work on subdomains if they are registered on the root domain**, so all passkeys registered on `domain.com` would work for any subdomains. For example:

- Passkeys created on `localhost` will only work on `localhost`.

> - Passkeys created on your Account Portal (e.g., `your-app.accounts.dev`) will only work on that domain.

Reviewer: ❓ What's the reason to remove these two examples?

Author: @mzhong9723 I think I'll prob re-add those under the explanations for a `development` environment linked to your comment [here](https://github.com/clerk/clerk-docs/pull/2291/files#r2140656206)

Author: Added this

> - Passkeys created on `staging1.your-domain.com` **cannot be used** on `staging2.your-domain.com` (sibling subdomains).

Reviewer 2: This is a much more clear explanation - I just want to make sure you're very confident this is true or have tested it?

Reviewer: This case is tricky - the important part is that the Clerk app's primary domain must be `your-domain.com` in order for passkeys to work on subdomains. So it's possible that a passkey _created on_ `staging1.your-domain.com` can be used for `staging2.your-domain.com` as long as `your-domain.com` is the primary domain for the app. 

Here's the backend logic for this: https://github.com/clerk/clerk_go/blob/8bfcac7105bd40e1348a438bbdbfc3a7390b2c24/api/shared/passkeys/service.go#L101.

Author: @mzhong9723 Yeah that's what I tried to convey with this sentence here: 

**However, passkeys **do work on subdomains if they are registered on the root domain**, so all passkeys registered on `domain.com` would work for any subdomains.**

To my understanding of the logic you linked above, if the app's primary domain was `staging1.your-domain.com`, then, passkeys created there would not work on `staging2.your-domain.com`, because the RpID would be too specific. But if the passkey is set on `your-domain.com`, then passkeys become usable across these subdomains.

So that last example stands but could update to: 

- Passkeys created on `staging1.your-domain.com` **cannot be used** on `staging2.your-domain.com` **as long as `your-domain.com` is the primary domain for the app**. (sibling subdomains).

> [!IMPORTANT]

> > This means that **[satellite domains](/docs/advanced-usage/satellite-domains) do not support passkeys** because they are considered different domains. If you plan to use satellite domains - in either development or production - you should avoid using passkeys, as they won't be portable between your primary domain and your satellite domains.

Reviewer 3: since the sentence below this callout is "If you're **not** using satellite domains", then the paragraph before that _shouldn't_ be a callout, it should be in the same flow of the document. so i'm removing the callout here, and ordering the information as

**If you're using [satellite domains](/docs/advanced-usage/satellite-domains)**,

If you're **not** using satellite domains:

```suggestion
**If you're using [satellite domains](/docs/advanced-usage/satellite-domains)**, in both development and production, they do not support passkeys because they are considered different domains. With satellite domains, you should avoid using passkeys as they won't be portable between your primary domain and your satellite domains.
```

Author: Makes sense! Thanks!

If you're **not** using satellite domains:

- **In development**, you can either:

>   - Use Clerk's [Components](/docs/components/overview), [Elements](/docs/customization/elements/overview), or [custom flows](/docs/custom-flows/overview), instead of the Account Portal. This ensures the passkey is created and used entirely on your development domain, so passkeys created on `localhost` will only work on `localhost`. **This is the recommended approach**

Reviewer 3: ```suggestion
  - **The recommended approach**. Use Clerk's [components](/docs/components/overview), [Elements](/docs/customization/elements/overview), or [custom flows](/docs/custom-flows/overview), instead of the [Account Portal](/docs/account-portal/overview). This ensures the passkey is created and used entirely on your development domain, so passkeys created on `localhost` will only work on `localhost`.
```

>   - Create the passkey directly through Account Portal instead of your local application to keep passkeys tied to the Account Portal's domain.

Reviewer 2: The way I'm reading this is "only use AP or never use AP if you want to use passkeys", is that right? If so, I wonder if we'd want to take a stronger stance here tbh. Exactly replicating AP with a custom page is like a 5 minute task, and feels like a much safer approach to me than locking yourself into AP indefinitely, when realistically most businesses end up moving over to a custom page/flow as they scale for customization/branding reasons.

Reviewer: Is the first bullet point just saying to use develop on `localhost`? I would expect this is what most developers would be doing anyways. I think the important thing to make clear is that passkeys can't be shared between localhost and development instance account portal, and there's no way around that.

Author: Added changes

>   - Create a passkey directly through the Account Portal instead of your local application to keep it tied to the portal's domain. Passkeys created on your Account Portal (e.g., `your-app.accounts.dev`) will only work on that domain, which can cause issues if you switch between `localhost` and the Account Portal during developemnt. If you choose this approach, ensure all testing happens on the same domain where the passkey was created.

Reviewer 3: ```suggestion
  - Create a passkey directly through the Account Portal instead of your local application to keep it tied to the Account Portal's domain. Passkeys created on your Account Portal (e.g., `your-app.accounts.dev`) will only work on that domain, which can cause issues if you switch between `localhost` and the Account Portal during development. If you choose this approach, ensure all testing happens on the same domain where the passkey was created.
```

> - **In production,** this limitation typically **does not apply** because your Account Portal is usually hosted on a subdomain of your main domain (e.g. `accounts.your-domain.com`), enabling passkeys to work seamlessly across your app. However, as stated above, if you use **satellite domains**, this issue **does apply** and it is not recommended to use passkeys in that scenario.

Reviewer 3: ```suggestion
- **In production,** your Account Portal is usually hosted on a subdomain of your main domain (e.g. `accounts.your-domain.com`), enabling passkeys to work seamlessly across your app. However, as stated above, if you use **satellite domains**, passkeys will not work as intended.
```
